### DISCUSSION TITLE
Intellisense on mac not working as specified in configurations

### DISCUSSION THREAD

**Discussion by lollospadalaser** [microsoft/vscode-cpptools#12751] [Q&A]
the command I use to compile my file is g++ -D 

-std=c++17 test.cpp -o test

on the terminal if I run: which g++ the output is:
/usr/bin/g++

I added the option the arguments and selected the g++ compiler in the intellisense configurations:
<img width="1293" alt="Screenshot 2024-09-22 at 17 34 49" src="https://github.com/user-attachments/assets/14dc088f-3cbc-4bea-8548-f259a9e14151">

but the intellisense isn't working correctly
<img width="1072" alt="Screenshot 2024-09-22 at 17 50 57" src="https://github.com/user-attachments/assets/a57b21df-92cd-4200-820f-3957ce117205">

the intellisense says that there isn't a suitable conversion but there is one because the code runs correctly.

this is the same for many compiler, for the g++-14 compiler (g++ installed through homebrew) some errors don't get detected by the intellisense, while they show up during the compilation

**Comment by nayfnaif55** [comment]
٢
